gerrit-checklist.js

```diff
@@ -79,82 +163,105 @@ ReviewChecklistManager.prototype.createStandinTextArea = function() {
 
 ReviewChecklistManager.prototype.updateManagedTextArea = function() {
   if (!this.gerTextArea) return;
-  this.gerTextArea.val('' + this.textArea.val() + this.checklistText());
+  this.gerTextArea.value = '' + this.textArea.value + this.checklistText();
+  this.gerTextArea.dispatchEvent(new Event('input', {bubbles: true}));
 };
 
 ReviewChecklistManager.prototype.insertOptions = function() {
-  this.textArea.after(this.optionsTable());
+  this.textArea.parentElement.appendChild(buildElement('hr'));
+  this.textArea.parentElement.appendChild(this.optionsTable());
 };
 
 ReviewChecklistManager.prototype.optionsTable = function() {
-  var opts = this.buildOptionsTable();
-  this.optionsTable = function() { return opts };
```

Comment:
whoa, self-replacing function, that's some elite ninja coding technique right there